src/plugins/mavlink_ftp/include/plugins/mavlink_ftp/mavlink_ftp.h

```diff
@@ -0,0 +1,220 @@
+#pragma once
+
+#include <vector>
+#include <memory>
+#include <string>
+#include <functional>
+
+// This plugin provides/includes the mavlink 2.0 header files.
+#include "mavlink/v2.0/common/mavlink.h"
+#include "plugin_base.h"
+
+namespace mavsdk {
+
+class System;
+class MavlinkFTPImpl;
+
+/**
+ * @brief The MavlinkFTP class provides MAVLink FTP functionality
+ *
+  */
+class MavlinkFTP : public PluginBase {
+public:
+    /**
+     * @brief Constructor. Creates the plugin for a specific System.
+     *
+     * The plugin is typically created as shown below:
+     *
+     *     ```cpp
+     *     auto mavlink_ftp = std::make_shared<MavlinkFTP>(system);
+     *     ```
+     *
+     * @param system The specific system associated with this plugin.
+     */
+    explicit MavlinkFTP(System &system);
+
+    /**
+     * @brief Destructor (internal use only).
+     */
+    ~MavlinkFTP();
+
+    /**
+     * @brief Possible results returned for requests.
+     */
+    enum Result : uint8_t {
+        kSuccess,
```

Comment:
We tend to write more [`SUCCESS`](https://github.com/mavlink/MAVSDK/blob/develop/src/plugins/mission/include/plugins/mission/mission.h#L46), etc. Which we should change eventually because it caused us issues with e.g. already defined `ERROR` on Windows, but well, that's how it is now.

Reply:
Changed to the MAVSDK coding style. Previous was copy/pasted from Mavlink FTP implementation in Firmware.